tests/rptest/tests/group_membership_test.py

```diff
@@ -235,12 +235,21 @@ def check_metric():
             assert metric_key in metrics_offsets
             assert metrics_offsets[metric_key] == 50
 
-            self.client().delete_topic(topic)
-
         check_metric()
+
+        self.client().delete_topic(topic)
+        wait_until(lambda: topic_spec.name not in rpk.list_topics(),
+                   timeout_sec=10,
+                   backoff_sec=1,
+                   err_msg="Topic is not deleted")
         metrics_offsets = self._get_offset_from_metrics(group)
         assert metrics_offsets is None
```

Comment:
@ZeDRoman this looks ok, but it seems like it could be simplified:

I'm guessing that you are using `list_topics` as a sort of guard proxy to signal that the change should now be reflected in the metrics. however, you can simplify this by testing for the condition you want _directly_, with something like:

```python
wait_until(lambda: self._get_offset_from_metrics(group)==None, timeout_sec=10, ...)
```

They achieve a similar result, but by using list_topics check a future reader of the code will have to derive for themselves why this list topics has anything to do with metrics.